packages/runtime/plugin-runtime/src/router/cli/handler.ts

```diff
@@ -0,0 +1,53 @@
+import path from 'path';
+import { Entrypoint } from '@modern-js/types';
+import { PluginAPI } from '@modern-js/core';
+import { AppTools } from '@modern-js/app-tools';
+import { cloneDeep } from '@modern-js/utils/lodash';
+import { modifyEntrypoints } from './entry';
+import { isPageComponentFile } from './code/utils';
+
+let originEntrypoints: any[] = [];
+
+export async function handleModifyEntrypoints(
+  api: PluginAPI<AppTools<'shared'>>,
+  entrypoints: Entrypoint[],
+) {
+  const newEntryPoints = modifyEntrypoints(entrypoints);
+  const appContext = api.useAppContext();
+  const resolvedConfig = api.useResolvedConfigContext();
+  appContext.entrypoints = newEntryPoints;
+  originEntrypoints = cloneDeep(newEntryPoints);
+  const { generateCode } = await import('./code');
```

Comment:
Perhaps in the future we should remove the generated code logic from `ModifyEntryPoints`..put in separate place

Reply:
yeap